### PR TITLE
grid_map_core: Use ament_export_targets and improve eigen linkage

### PR DESCRIPTION
**Pull request by Ryanf55**
This fixes include errors in grid_map_geo ros2 port. If you need, I can rebase and target to rolling first.
Relates to #403, but to close that issue out, these changes should ideally be reflected across the whole repo.

### CODE REVIEW

File: grid_map_cv/CMakeLists.txt [hunk 1]
```diff
@@ -43,10 +43,14 @@ add_library(${PROJECT_NAME} SHARED
   src/InpaintFilter.cpp
 )
 
-ament_target_dependencies(${PROJECT_NAME} SYSTEM
+ament_target_dependencies(${PROJECT_NAME} PUBLIC
   ${dependencies}
 )
 
+target_link_libraries(${PROJECT_NAME} PUBLIC
+  grid_map_core::grid_map_core
```
Comment (wep21) [inline grid_map_cv/CMakeLists.txt:51]: How about adding other dependencies like `rclcpp::rclcpp` and removing `ament_target_dependencies`?

Comment (Ryanf55) [reply grid_map_cv/CMakeLists.txt:51]: Yes, with more time, it would be a great idea to do that. Do you have ability to help out?

Comment (Ryanf55) [reply grid_map_cv/CMakeLists.txt:51]: I could split this PR to only touch `grid_map_core`, and then do one package at a time.

Comment (wep21) [reply grid_map_cv/CMakeLists.txt:51]: @Ryanf55 
> Do you have ability to help out?
Yes, I will work on it after this PR is once merged.

Comment (wep21) [reply grid_map_cv/CMakeLists.txt:51]: Here is [the PR](https://github.com/Ryanf55/grid_map/pull/1)

**Comment by wep21** [comment]
@Ryanf55 This PR includes some commits for rolling. Is it better to change the target branch to rolling?

**Comment by Ryanf55** [comment]
> @Ryanf55 This PR includes some commits for rolling. Is it better to change the target branch to rolling?

Ah yes, I've changed the target branch.

**Comment by Ryanf55** [comment]
@mergifyio backport iron

**Comment by Ryanf55** [comment]
@mergifyio backport humble

**Pull request merged**